src/reactor/fns.js

```diff
@@ -434,11 +469,53 @@ function cacheValue(reactorState, getter, value) {
     })
   })
 
-  return reactorState.setIn(['cache', cacheKey], Immutable.Map({
-    value: value,
-    storeStates: storeStates,
-    dispatchId: dispatchId,
-  }))
+  const maxItemsToCache = reactorState.get('maxItemsToCache')
+  const itemsToCache = maxItemsToCache * CACHE_CLEAR_RATIO
+
+  return reactorState.withMutations(state => {
+    if (maxItemsToCache && maxItemsToCache <= state.get('cache').size) {
+      do {
+        let key = state.get('cacheRecency').first()
+        state.deleteIn(['cache', key])
+        state.deleteIn(['cacheRecency', key])
+      } while (itemsToCache < state.get('cache').size)
+    }
+
+    state.setIn(['cacheRecency', cacheKey], cacheKey)
+    state.setIn(['cache', cacheKey], Immutable.Map({
+      value: value,
+      storeStates: storeStates,
+      dispatchId: dispatchId,
+    }))
+  })
+}
+
+/**
+ * Remove getter cache value from cache and recency cache
+ * @param {ReactorState} reactorState
+ * @param {getter} getter
+ * @return {ReactorState}
+ */
+function removeCacheValue(reactorState, getter) {
+  const cacheKey = getCacheKey(getter)
+  return reactorState.withMutations(rState => {
+    rState.deleteIn(['cache', cacheKey])
+    rState.deleteIn(['cacheRecency', cacheKey])
+  })
+}
+
+/**
+ * Readds the key for the item in cache to update recency
+ * @param {ReactorState} reactorState
+ * @param {getter} getter
+ * @return {ReactorState}
+ */
+function updateCacheRecency(reactorState, getter) {
+  const cacheKey = getCacheKey(getter)
+  return reactorState.withMutations(state => {
+    state.deleteIn(['cacheRecency', cacheKey])
```

Comment:
This `deleteIn()` isn't needed. `setIn()` is called for same key path directly after. 

```js
return reactorState.setIn(['cacheRecency', cacheKey], cacheKey)
```